### PR TITLE
[ci] move windows train tests to rayci

### PR DESCRIPTION
**Pull request by can-anyscale**
Move windows train tests to the common rayci tool. This requires the following changes:
- Turn off virus scanning in windows, this slow down docker build significantly
- Add the `--operating-system` flag to rayci to have a few special logic for windows (added `--os` tag originally but the word `os` is ambiguous to another library in python)
- Clean up build artifacts aggressively to speed up docker build

Test:
- CI

### CODE REVIEW

**Comment by can-anyscale** [comment]
docker build in windows is significantly slower than linux when creating layers: https://buildkite.com/ray-project/premerge/builds/13523#018c315f-55ca-4c59-8912-5561e499468e/6-5799 (20 minutes after ray build). I'm still investigating this, but this problem will alleviate for the most part when we port wanda to windows

File: ci/ray_ci/tests.windows.env.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+FROM rayproject/buildenv:windows
```
Comment (aslonnie) [inline ci/ray_ci/tests.windows.env.Dockerfile:1]: could you make this filename shorter? why does it need to have 4 words?

maybe create a directory just for windows under `ray_ci` or something?

File: ci/ray_ci/tester_container.py [hunk 2]
```diff
@@ -15,6 +15,7 @@ class TesterContainer(Container):
     def __init__(
         self,
         docker_tag: str,
+        operating_system: str = "linux",
```
Comment (aslonnie) [inline ci/ray_ci/tester_container.py:18]: please use `Optional[str] = None`, and if it is None, set it to "linux".

Otherwise, this `"linux"` will be initialized as a global valuable, and if its content is mutated in the function, it will have unintended mysterious effects.

File: ci/ray_ci/tester.py [hunk 3]
```diff
@@ -288,6 +300,8 @@ def _get_test_targets(
         .strip()
         .split("\n")
     )
+    # remove additional trailing whitespace on windows OS
+    test_targets = {t.rstrip() for t in test_targets}
```
Comment (aslonnie) [inline ci/ray_ci/tester.py:304]: how about changing line 300 to `split("\n")` and then `strip()`

the reason that you need this is because line break are 2 chars on windows. in fact, you can use `os.linesep` on the splitting: 

https://docs.python.org/3/library/os.html#os.linesep

File: ci/ray_ci/tester.py [hunk 4]
```diff
@@ -154,7 +161,9 @@ def main(
     if not bazel_workspace_dir:
         raise Exception("Please use `bazelisk run //ci/ray_ci`")
     os.chdir(bazel_workspace_dir)
-    docker_login(_DOCKER_ECR_REPO.split("/")[0])
+    # TODO: remove the if condition when wanda is supported for windows
+    if operating_system == "linux":
+        docker_login(_DOCKER_ECR_REPO.split("/")[0])
```
Comment (aslonnie) [inline ci/ray_ci/tester.py:166]: this has not much to do with wanda, this is just AWS ec2 machine credentials.

wanda is the container image building tool.

File: ci/ray_ci/test_tester.py [hunk 5]
```diff
@@ -35,7 +35,7 @@ def test_get_container() -> None:
         "ci.ray_ci.tester_container.TesterContainer.install_ray",
         return_value=None,
     ):
-        container = _get_container("core", 3, 1, 2, 0)
+        container = _get_container("core", 3, 1, 2, "linux", 0)
```
Comment (aslonnie) [inline ci/ray_ci/test_tester.py:38]: this has so many args now, could you add arg keys here to make is more readable?

File: ci/ray_ci/requirements.in [hunk 6]
```diff
@@ -1,3 +1,4 @@
 click
+colorama
```
Comment (aslonnie) [inline ci/ray_ci/requirements.in:2]: why is this required now?

colorama historically has introduced issues because of its markdown related dependencies.

Comment (can-anyscale) [reply ci/ray_ci/requirements.in:2]: running the ray_ci tool in windows complaining about this missing dependencies; i can make it a window-specific dependency

File: ci/ray_ci/tester.py (hunk 3 above)
Comment (can-anyscale) [reply ci/ray_ci/tester.py:304]: #verynice

**Comment by can-anyscale** [comment]
Let me break this in multiple PR to introduce sub-classes
